cpp/src/arrow/flight/types.cc

```diff
@@ -234,6 +260,23 @@ Status FlightInfo::Deserialize(const std::string& serialized,
   return Status::OK();
 }
 
+Status FlightInfo::Deserialize(const Buffer& serialized,
+                               std::unique_ptr<FlightInfo>* out) {
```

Comment:
Nit: Is it possible to combine these XXX:Deserialize functions into one, if change this unique_ptr to raw pointer as other functions do.